src/hotspot/cpu/arm/c1_LIRAssembler_arm.cpp

```diff
@@ -1820,8 +1820,8 @@ void LIR_Assembler::comp_op(LIR_Condition condition, LIR_Opr opr1, LIR_Opr opr2,
         __ teq(xhi, yhi);
         __ teq(xlo, ylo, eq);
       } else {
-        __ subs(xlo, xlo, ylo);
-        __ sbcs(xhi, xhi, yhi);
+        __ cmp(xlo, ylo);
```

Comment:
nit: we may want to use `__ subs(Rtemp, xlo, ylo);` here to align with the usage in match rules in arm.ad, e.g., `compL_reg_reg_LEGT`. It's okay to me if you use `cmp` anyway.

Reply:
I agree.

Reply:
Aligned with C2  implementation